docs/config-prereqs.txt

```diff
@@ -1181,9 +1181,9 @@ systems, can add dash-number suffixed symlinks to compiler tools (e.g.
 `gcc-7` beside the `gcc` installed by package) near the original
 binaries and into `/usr/lib/ccache`:
 ------
-:; ( cd /usr/bin && for TOOL in cpp gcc g++ ; do \
-        for VER in "-7" "-14" ; do \
-            ln -s "$TOOL" "$TOOL$VER" ; \
+:; ( cd /usr/pkg/bin && for TOOL in cpp gcc g++ ; do \
+        for VER in "7" "14" ; do \
+            ln -fs "../gcc$VER/bin/$TOOL" "${TOOL}-${VER}" ; \
```

Comment:
This suggests you are running into "nut can't just build with the system compiler", which is gcc7 on NetBSD 9, gcc10 on 10, and gcc12 on -current.

Or is this "intentionally do builds with different compilers to provoke different kinds of warnings, to check them all"?  In which case, nevermind, makes sense.

Reply:
Yes, the idea is that https://github.com/networkupstools/jenkins-dynamatrix and https://github.com/networkupstools/nut/blob/master/Jenkinsfile-dynamatrix conspire to dynamically define a build matrix and cover different toolkits, with default scripts expecting patterns like `${COMPILER}-${COMPILERVER}` where versions are iterated. With known compiler versions, we can then automatically iterate a number of C standards known supported by the toolkit release.

By the silent default, it is up to autotools to find something (which is typically some `gcc` first seen via the `PATH`).